### ISSUE TITLE
write_prometheus and instance in different timezone

### ISSUE THREAD

**Issue by petarkozic** [collectd/collectd#2718]
*   Version of collectd:
collectd 5.8.0.63.ga68a482

*   Operating system / distribution:
Debian 8 Jessie

I have one instance in different time zone than prometheus. On that instance I'm using collectd with write_prometheus.
On prometheus log I can see: msg="Error on ingesting samples that are too old or are too far into the future"
I supposed, that is because that instance was in different time zone but another problem is becuase collectd write timestamp in export.

Can I disable timestamp or what can I do to avoid this problem ?

**Comment by tagGeeY** [comment]
I also ran into this today.

**Comment by eero-t** [comment]
Metrics need to be timestamped at source, otherwise metrics from different sources can't be accurately correlated.  As a corollary, all your nodes needs to use reliable (e.g. NTP) clock source, regardless of where they are.

Is your host clock accurate and have you set correct timezone for Collectd process / container in regards to what your host machine clock uses?

(There are some embedded devices which may not have accurate enough clock, I guess in that case one needs to use a proxy for collecting the data which timestamps the data. I'm not a Collectd developer, so I have no idea whether it provides any support for something like that.)

**Comment by tagGeeY** [comment]
> Metrics need to be timestamped at source, otherwise metrics from different sources can't be accurately correlated. As a corollary, all your nodes needs to use reliable (e.g. NTP) clock source, regardless of where they are.
> 
> Is your host clock accurate and have you set correct timezone for Collectd process / container in regards to what your host machine clock uses?
> 
> (There are some embedded devices which may not have accurate enough clock, I guess in that case one needs to use a proxy for collecting the data which timestamps the data. I'm not a Collectd developer, so I have no idea whether it provides any support for something like that.)

Yes... The host was set wrong time zone, setting the correct time zone worked for me.

**Comment by mrunge** [comment]
Right. prometheus adds a timestamp at sampling time; but there is an option to also use the timestamp from source, and that is what happens here.
